clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -299,17 +299,28 @@ public void ensurePartitionAssignment() {
             ensureActiveGroup();
     }
 
-    @Override
-    public void close() {
-        // commit offsets prior to closing if auto-commit enabled
-        while (true) {
-            try {
-                maybeAutoCommitOffsetsSync();
-                return;
-            } catch (WakeupException e) {
-                // ignore wakeups while closing to ensure we have a chance to commit
-                continue;
+    public void close(long timeoutMs) {
+        if (timeoutMs < 0)
+            throw new IllegalArgumentException("Timeout must be non-negative");
+
+        long now = time.milliseconds();
+        long deadline = now + timeoutMs;
+        if (deadline < 0) deadline = Long.MAX_VALUE;
```

Comment:
That's also what I had in mind initially, but it seemed like only a little more effort to give the user full control over shutdown, which I thought would be appreciated. That said, I'm also ok with the simpler blocking approach since, as you point out, it is bounded by the request timeout.